pkg/reconciler/common/images.go

```diff
@@ -41,34 +41,80 @@ var (
 	delimiter             = "/"
 )
 
+// ImageTransformer is an interface for transforming images passed to the ResourceImageTransformer
+type ImageTransformer interface {
+	HandleImage(containerName, parent string, useDefault bool) (string, bool)
+	HandleImagePullSecrets(imagePullSecrets []corev1.LocalObjectReference, log *zap.SugaredLogger) []corev1.LocalObjectReference
+}
+
+// registryImageTransformer is a v1alpha1.Registry specific transformer
+type registryImageTransformer struct {
+	registry *v1alpha1.Registry
+}
+
+var _ ImageTransformer = (*registryImageTransformer)(nil)
+
+func (rit *registryImageTransformer) HandleImage(containerName, parent string, useDefault bool) (string, bool) {
+	if image, ok := rit.registry.Override[parent+delimiter+containerName]; ok {
+		return image, true
+	}
+	if image, ok := rit.registry.Override[containerName]; ok {
+		return image, true
+	}
+	if !useDefault {
+		return "", false
+	}
+	return replaceName(rit.registry.Default, containerName), true
+}
+
+func replaceName(imageTemplate string, name string) string {
+	return strings.ReplaceAll(imageTemplate, containerNameVariable, name)
+}
+
+func (rit *registryImageTransformer) HandleImagePullSecrets(imagePullSecrets []corev1.LocalObjectReference, log *zap.SugaredLogger) []corev1.LocalObjectReference {
+	if len(rit.registry.ImagePullSecrets) > 0 {
+		log.Debugf("Adding ImagePullSecrets: %v", rit.registry.ImagePullSecrets)
+		imagePullSecrets = append(imagePullSecrets, rit.registry.ImagePullSecrets...)
+	}
+	return imagePullSecrets
+}
+
 // ImageTransform updates image with a new registry and tag
 func ImageTransform(registry *v1alpha1.Registry, log *zap.SugaredLogger) mf.Transformer {
+	rit := &registryImageTransformer{
+		registry: registry,
+	}
+	return ResourceImageTransformer(rit, log)
+}
+
+// ResourceImageTransformer takes an ImageTransformer and transform images across resources
+func ResourceImageTransformer(imageTransformer ImageTransformer, log *zap.SugaredLogger) mf.Transformer {
```

Comment:
why public?

Reply:
Making this public is the point of the change. It allows for other people to make an `ImageTransformer` and pass it to `ResourceImageTransformer`.